### PR TITLE
[plot] misc bug fixes

### PR DESCRIPTION
**Pull request by larkin**
Includes bug fixes for #2025, #2024, #2014.  Also removes an unused file.

# Author Checklist

1. Changes address original issue? Y
2. Unit tests included and/or updated with changes? N, issue exists for backfill.
3. Command line build passes? Y
4. Changes have been smoke-tested? Y

### CODE REVIEW

File: src/plugins/plot/src/configuration/YAxisModel.js [hunk 1]
```diff
@@ -146,10 +146,14 @@ define([
...
                 }
             }, this);
+            this.listenTo(series, 'change:yKey', function () {
+                this.updateFromSeries(this.seriesCollection);
+            }, this);
         },
         untrackSeries: function (series) {
             this.stopListening(series);
             this.resetStats();
+            this.updateFromSeries(this.seriesCollection);
```
Comment (ev1stensberg) [inline src/plugins/plot/src/configuration/YAxisModel.js]: #2029 might be relative to fix the issue it is closing. `this.updateFromSeries` doesn't get called when telemetry is empty / table is empty. The function sets defaults that are used later for empty telemetry tables.

**Comment by larkin** [comment]
@psarram idea is if a user customized, they don't want the customization removed just because they removed a series.  e.g. they accidentally added the incorrect telemetry point and then remove.

**Comment by larkin** [comment]
Have fixed the bug with the selected y-axis value not being shown in edit mode.

Good for re-review and merge.

File: src/plugins/plot/res/templates/plot-options-edit.html [hunk 2]
```diff
@@ -46,12 +46,13 @@
                             <label>Value:</label>
                             <span class="control">
                                 <div class="select">
-                                    <select ng-model="form.series[$index].yKey">
-                                        <option ng-repeat="option in form.series[$index].yAxisOptions"
+                                    <select ng-model="seriesForm.yKey">
+                                        <option ng-repeat="option in seriesForm.yAxisOptions"
                                                 value="{{option.value}}"
-                                                ng-selected="option.value == form.series[$index].yKey">
+                                                ng-selected="option.value === seriesForm.yKey">
```
Comment (ev1stensberg) [inline src/plugins/plot/res/templates/plot-options-edit.html]: A less strict equal here could be of use if the data somehow changes value, but strict equal is 👍

**Comment by larkin** [comment]
Rebased this to fix conflicts after merging plot ui fixes, good for review and integrate.

**Comment by dtailor90** [comment]
Looks good. Verified issues it fixes. 

## Reviewer Checklist
Changes appear to address issue? Y
Appropriate unit tests included? N/A, issue exists for backfill.
Code style and in-line documentation are appropriate? Y
Commit messages meet standards? Y

**Pull request merged**